warehouse/query-core/src/main/java/datawave/query/jexl/visitors/AddShardsAndDaysVisitor.java

```diff
@@ -0,0 +1,120 @@
+        // If the root node is an ASTJexlScript, go down one level before adding the SHARDS_AND_DAYS node.
+        if (node instanceof ASTJexlScript) {
+            addNewShardAndDaysNode(node.jjtGetChild(0), shardsAndDays);
+        } else {
+            // Otherwise, add the SHARDS_AND_DAYS node.
+            addNewShardsAndDaysNode(node, shardsAndDays);
+        }
+    }
+    
+    // Create and append a new SHARDS_AND_DAYS hint node and AND it to the given node.
+    private static void addNewShardsAndDaysNode(JexlNode node, String shardsAndDays) {
+        JexlNode originalParent = node.jjtGetParent();
+        JexlNode shardAndHintNode = JexlNodes.wrap(JexlNodeFactory.createAssignment(Constants.SHARD_DAY_HINT, shardsAndDays));
+        // If the current node is unwrapped, wrap it before ANDing it with the new SHARDS_AND_DAYS node.
+        JexlNode leftSide = !(node instanceof ASTReference) ? JexlNodes.wrap(node) : node;
+        JexlNode andNode = JexlNodeFactory.createAndNode(Lists.newArrayList(leftSide, shardAndHintNode));
+        // Update the grandparent's lineage.
+        if (originalParent != null) {
+            JexlNodes.replaceChild(originalParent, node, andNode);
+        }
+    }
+    
+    private final List<String> shardsAndDays;
+    private boolean updatedShardsAndDays = false;
+    
+    private AddShardsAndDaysVisitor(List<String> shardsAndDays) {
+        this.shardsAndDays = shardsAndDays;
+    }
+    
+    @Override
+    public Object visit(ASTAssignment node, Object data) {
+        ASTAssignment copy = (ASTAssignment) super.visit(node, data);
+        String identifier = JexlASTHelper.getIdentifier(copy);
+        // If this is a SHARDS_AND_DAYS hint node, append the new shards and days hints to its literal value. Only do this to the first SHARDS_AND_DAYS node
+        // that is found, if any.
+        if (Constants.SHARD_DAY_HINT.equals(identifier) && !updatedShardsAndDays) {
+            String nodeShardsAndDays = JexlASTHelper.getLiteralValue(copy).toString();
+            String mergedNodeShardsAndDays = mergeShardsAndDays(nodeShardsAndDays);
+            this.updatedShardsAndDays = true;
+            return JexlNodeFactory.createAssignment(Constants.SHARD_DAY_HINT, mergedNodeShardsAndDays);
+        }
+        return copy;
+    }
+    
+    // Merge the given shards and days with the shard and days in this visitor's list.
+    private String mergeShardsAndDays(String shardsAndDays) {
+        // Remove any duplicate shards and days hints from the initial list.
+        String[] values = StringUtils.split(shardsAndDays, ',');
+        for (String value : values) {
+            this.shardsAndDays.remove(value);
+        }
+        // If they were all duplicates, return the original string. Otherwise append them.
+        if (this.shardsAndDays.isEmpty()) {
+            return shardsAndDays;
+        } else {
+            return shardsAndDays + "," + JOINER.join(this.shardsAndDays);
```

Comment:
I know it is an edge case and perhaps will never happen, but if the shardsAndDays is the empty string, then this will result is a leading comma.

Reply:
Good catch, I have added a fix to handle this edge case.